src/services/hooks/use-token-balance.ts

```diff
@@ -0,0 +1,82 @@
+import { ChainBalance, CurrencyUnit, GovernanceUnit } from '@interlay/interbtc-api';
+import { Currency } from '@interlay/monetary-js';
+import { useErrorHandler } from 'react-error-boundary';
+import { useQuery, useQueryClient, UseQueryResult } from 'react-query';
+import { useSelector } from 'react-redux';
+
+import { StoreType } from '@/common/types/util.types';
+import { GOVERNANCE_TOKEN } from '@/config/relay-chains';
+import genericFetcher, { GENERIC_FETCHER } from '@/services/fetchers/generic-fetcher';
+import useAccountId from '@/utils/hooks/use-account-id';
+
+interface UseTokenBalance<T extends CurrencyUnit> {
+  tokenBalanceIdle: UseQueryResult<ChainBalance<T>, Error>['isIdle'];
+  tokenBalanceLoading: UseQueryResult<ChainBalance<T>, Error>['isLoading'];
+  tokenBalance: UseQueryResult<ChainBalance<T>, Error>['data'];
+}
+
+const useTokenBalance = <T extends CurrencyUnit>(
+  token: Currency<T>,
+  accountAddress: string | undefined
+): UseTokenBalance<T> => {
+  const { bridgeLoaded } = useSelector((state: StoreType) => state.general);
+
+  const accountId = useAccountId(accountAddress);
+
+  const {
+    isIdle: tokenBalanceIdle,
+    isLoading: tokenBalanceLoading,
+    data: tokenBalance,
+    error: tokenBalanceError
+  } = useQuery<ChainBalance<T>, Error>(
+    [GENERIC_FETCHER, 'tokens', 'balance', token, accountId],
```

Comment:
Not for now, but a we're moving towards of using hooks for api calls, I'd like to move away from the generic fetcher (since we'll only need to specify the api url and query key once, in the hook). Not a blocker for this PR, but I've added a ticket to sprint 43.

Reply:
OK, let me take care of it in that sprint.